pkg/controller/admissionchecks/multikueue/multikueuecluster.go

```diff
@@ -98,30 +122,63 @@ func (rc *remoteClient) setConfig(watchCtx context.Context, kubeconfig []byte) e
 	}
 	rc.client = remoteClient
 
-	newWatcher, err := rc.client.Watch(watchCtx, &kueue.WorkloadList{})
+	err = rc.startWatcher(watchCtx, kueue.GroupVersion.WithKind("Workload").GroupKind().String(), &workloadKueueWatcher{})
 	if err != nil {
-		return nil
+		return err
+	}
+
+	// add a watch for all the adapters implementing multiKueueWatcher
+	for kind, adapter := range adapters {
+		watcher, implementsWatcher := adapter.(multiKueueWatcher)
+		if !implementsWatcher {
+			continue
+		}
+		err := rc.startWatcher(watchCtx, kind, watcher, client.HasLabels{constants.PrebuiltWorkloadLabel})
+		if err != nil {
+			// not being able to setup a watcher is not ideal but we can function
+			// with only thw wl watcher, just log the error.
+			ctrl.LoggerFrom(watchCtx).V(2).Error(err, "unaable to start the watcher", "kind", kind)
+		}
+	}
+
+	rc.kubeconfig = kubeconfig
+	return nil
+}
+
+func (rc *remoteClient) startWatcher(ctx context.Context, kind string, w multiKueueWatcher, listopts ...client.ListOption) error {
+	log := ctrl.LoggerFrom(ctx).WithValues("watchKind", kind)
+	newWatcher, err := rc.client.Watch(ctx, w.GetEmptyList())
+	if err != nil {
+		return err
 	}
 
 	go func() {
+		log.V(3).Info("Starting watch")
+		defer log.V(3).Info("Watch ended")
 		for r := range newWatcher.ResultChan() {
-			rc.queueWorkloadEvent(watchCtx, r)
+			wlKey, err := w.GetEventsWorkloadKey(r.Object)
+			if err != nil {
+				log.V(2).Error(err, "Cannot get workload key")
```

Comment:
Log the namespace/name for the object for which we got the error, so that we can identify it

Reply:
`r.Object` is a runtime object and is not exposing the name, however I made it part of the error message for JobSet.

Reply:
Ok, it is acceptable if there is no easy way to know the name / namespace of the object here. 

However, if possible I prefer to log it here so that it does not need to be done per adapter. Let see what are the options to make it generic.

Reply:
I guess we could at least log its gvk in a generic way.
Maybe we could also cast it on a Object containing name / namespace

Reply:
gvk added